### PR TITLE
PIM-5625: simplify InstallBundle

### PR DESCRIPTION
**Pull request by nidup**
**Description (for Contributor and Core Developer)**

In v1.5, InstallBundle contains 3 "generations" of code to import different objects,
 - 1) use doctrine fixtures to install data (locales for instance)
 - 2) use deprecated import (<= v1.3) to install data (channels for instance)
 - 3) use new import system (>=1.4) to install data (products for instance) 

These classes are used to install the PIM but also to install a Behat catalog (in a different way...) and to create objects in behat fixtures (using old or new API processor...).

The goal of this PR is to simplify the install of catalogs for the PIM and to simplify the Behat stack by using everywhere the same API to setup data :tada: 

A first step has been done to only use the new import system and not anymore the deprecated one in the context of the Installer (has been done with this PR https://github.com/akeneo/pim-community-dev/pull/4213)

This PR will allow to,
 - [x] Use only `akeneo:batch:job` to install all PIM data (remove useless doctrine fixtures layer)
 - [x] Use `akeneo:batch:job` to install a Behat catalog in CatalogConfigurationContext::aCatalogConfiguration() (don't use custom loader + registry and drop the related conf)
 - [x] Use processors and savers to load objects to replace FixturesContext::loadFixture()
 - [x] Enforce the fact that all files must be present in a installer_data set to avoid any installation weirdness (in other terms, if you don't need variant_group.csv, you must add an empty variant_group.csv file)

No impact on the install time, cf following measures on Community Edition with MongoDB storage and icecat_demo_dev fixtures (prod mode, no xdebug), before: real 2m4.797s, after: real 1m58.552s.

NB: this change will also speed up the work on the batch simplification https://github.com/akeneo/pim-community-dev/pull/4326 by using the v1.4 API everywhere.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | N
| Changelog updated                 | Y
| Review and 2 GTM                  | -
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

**Comment by jmleroux** [comment]
@nidup burning fever ?

:)

**Comment by nidup** [comment]
@jmleroux holidays, more cleanup to come :wink:

File: src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php [hunk 1]
```diff
@@ -116,6 +116,27 @@ public function deleteJobs()
     }
 
     /**
+     * Get the list of stored jobs
+     *
+     * @return JobInstance[]
+     */
+    public function getRunnableJobInstances()
```
Comment (pierallard) [inline src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php]: IMHO we should not do this here, and add a warning in the output when you run the job instance.
When you develop, you want to have a message "file product.csv not found" (when you have a "products.csv" file instead: ragequit. It appends to me 1000 times).

Comment (nidup) [inline src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php]: Yes, I'm trying to use the same code to install Behat catalog which does not contain all files, but we should enforce the fact that all files are presents to make it more predictible.

File: features/Context/CatalogConfigurationContext.php [hunk 2]
```diff
@@ -73,27 +67,33 @@ public function aCatalogConfiguration($catalog)
...
-            $this->runLoader($loader, $file);
+        // load JobInstances
+        $this->getFixtureJobLoader()->load($files);
+
+        // setup akeneo:batch:job command
+        $application = new Application();
+        $application->add(new BatchCommand());
+        $batchJobCommand = $application->find('akeneo:batch:job');
+        $batchJobCommand->setContainer($this->getContainer());
+        $commandTester = new CommandTester($batchJobCommand);
```
Comment (juliensnz) [inline features/Context/CatalogConfigurationContext.php]: $command :)

File: features/Context/CatalogConfigurationContext.php [hunk 3]
```diff
@@ -73,27 +67,33 @@ public function aCatalogConfiguration($catalog)
...
-            if ($file) {
-                $treatedFiles[] = $file;
-            }
-            $this->runLoader($loader, $file);
+        // load JobInstances
+        $this->getFixtureJobLoader()->load($files);
+
+        // setup akeneo:batch:job command
+        $application = new Application();
+        $application->add(new BatchCommand());
```
Comment (juliensnz) [inline features/Context/CatalogConfigurationContext.php]: maybe a good argument to have command as a service :)

File: src/Akeneo/Bundle/BatchBundle/Resources/config/removers.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+services:
+    akeneo_batch.remover.job_instance:
+        class: %akeneo_storage_utils.remover.base.class%
+        arguments:
+            - '@doctrine.orm.entity_manager'
+            - '@akeneo_storage_utils.remover.base_options_resolver'
+            - '@event_dispatcher'
+            - %akeneo_batch.entity.job_instance.class%
```
Comment (juliensnz) [inline src/Akeneo/Bundle/BatchBundle/Resources/config/removers.yml]: new yaml specification ask for ' around every argument.

**Comment by juliensnz** [comment]
Little comment, then GTM :)

File: src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php [hunk 5]
```diff
@@ -19,128 +20,112 @@ class FixtureJobLoader
...
+     * @param JobInstance[] $jobInstances
+     * @param array         $replacePaths
+     * @throws \Exception
+     * @return JobInstance[]
+     */
+    protected function configureJobInstances(array $jobInstances, array $replacePaths)
+    {
+        if (0 === count($replacePaths)) {
+            return $this->jobInstancesConfigurator->configureJobInstancesWithInstallerData($jobInstances);
         } else {
```
Comment (aRn0D) [inline src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php]: useless else

File: src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php [hunk 6]
```diff
@@ -19,128 +20,112 @@ class FixtureJobLoader
...
 
-        if (preg_match('/^(?P<bundle>\w+):(?P<directory>\w+)$/', $installerData, $matches)) {
-            $bundles = $this->container->getParameter('kernel.bundles');
-            $reflection = new \ReflectionClass($bundles[$matches['bundle']]);
-            $installerDataDir = dirname($reflection->getFilename()) . '/Resources/fixtures/' . $matches['directory'];
+        return $jobs;
+    }
+
+    /**
+     * @param JobInstance[] $jobInstances
```
Comment (aRn0D) [inline src/Pim/Bundle/InstallerBundle/FixtureLoader/FixtureJobLoader.php]: wrong alignment

File: src/Pim/Bundle/InstallerBundle/FixtureLoader/JobInstancesConfigurator.php [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    public function __construct(FixturePathProvider $pathProvider)
+    {
+        $this->pathProvider = $pathProvider;
+    }
+
+    /**
+     * The standard method to configure job instances with files provided in an install fixtures set
+     *
+     * @param JobInstance[] $jobInstances
+     * @throws \Exception
```
Comment (aRn0D) [inline src/Pim/Bundle/InstallerBundle/FixtureLoader/JobInstancesConfigurator.php]: empty line

**Pull request merged**